ast/compile.go

```diff
@@ -383,6 +384,14 @@ func (c *Compiler) WithPathConflictsCheck(fn func([]string) (bool, error)) *Comp
 	return c
 }
 
+// WithPathConflictsCheckRoot enables checking path conflicts from the specified root instead
+// of the top root node. This would enable optimizting path conflict checks during bundle
+// activation if a bundle already defines its own root paths.
+func (c *Compiler) WithPathConflictsCheckRoot(rootPaths []string) *Compiler {
```

Comment:
Any reason to use singular "root" for the name while the args and the property says "roots"?

Reply:
Good catch! Will change to use the plural form.